### PR TITLE
Create script for convert smart convert to byte code

### PR DESCRIPTION
**Pull request by ersul4ik**
Issue #34 

For further work, it is necessary to use the bytecode of a smart contract. 
The script needs the keys:

- --type . Which type you needed after convertation. ( Now we use only byte )
- --file . Which file do you need to convert. On moment script work without path. The file must be in the same directory as the script. It's decided if we have a contract name (for contract_interface)

### CODE REVIEW

File: generate_byte_code.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import click
```
Comment (s0b0lev) [inline generate_byte_code.py]: Did you add `click` to requirements.txt?

File: generate_byte_code.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    compiled_sol = compile_source(path_to_file.read())
+    contract_interface = compiled_sol['<stdin>:{}'.format(str(file.split('.sol')[0]))]
+    print(contract_interface['bin'])
+
+
+@click.command()
+@click.option('--type', help="format to convert")
+@click.option('--file', help="file to convert")
+def main(type, file):
+    if type == 'byte':
```
Comment (s0b0lev) [inline generate_byte_code.py]: Rename to bytecode

File: generate_byte_code.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+def convert_to_byte_code(file):
+    path_to_file = open(file, mode='r')
+    compiled_sol = compile_source(path_to_file.read())
+    contract_interface = compiled_sol['<stdin>:{}'.format(str(file.split('.sol')[0]))]
+    print(contract_interface['bin'])
+
+
+@click.command()
+@click.option('--type', help="format to convert")
```
Comment (s0b0lev) [inline generate_byte_code.py]: - [ ] `-t` flag support, 
- [ ] Define default parameter (bytecode and abi)
- [ ] Provide list of allowed parameters (bytecode, abi)

File: generate_byte_code.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@click.option('--file', help="file to convert")
```
Comment (s0b0lev) [inline generate_byte_code.py]: - [ ] Rename `--file` to `-s` `--solidity-file`
- [ ] Change help text to Uppercase : ) `Path to solidity file`
- [ ] Add `-c` `--contract` param, this parameter define which contract bytecode we need (if empty, provide dict with all contracts and bytecodes)

```
{
    'ContractName': {
        'bytecode': '0x..',
        'abi': '..'
    },
    ContractName2': {
        'bytecode': '0x..',
        'abi': '..'
    }
}

File: generate_byte_code.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+def main(type, file):
+    if type == 'byte':
+        convert_to_byte_code(file)
+
```
Comment (s0b0lev) [inline generate_byte_code.py]: - [ ] Provide  default behaviour
- [ ] Add unittest, provide test contract and pre-define bytecode/abi code

File: requirements.txt [hunk 6]
```diff
@@ -76,3 +76,4 @@ anybadge==1.1.1
 aiohttp-jinja2==1.0.0
 aioredis==1.1.0
 pycryptodome==3.6.6
+click==6.
```
Comment (s0b0lev) [inline requirements.txt]: Is it right version `6.`?

File: scanner/generate_byte_code.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    else:
+        contract_interface = compiled_sol['<stdin>:{}'.format(str(contract_name))]
+        report[str(contract_name)] = {'bytecode': contract_interface['bin'], 'abi': compiled_sol['<stdin>:' +
+                                                                                                 str(contract_name)]
+                                                                                                ['abi']}
+        print(report)
+
+
+def get_contract_names(file):
+    path_to_file = open(file, mode='r')
```
Comment (s0b0lev) [inline scanner/generate_byte_code.py]: You didnt close connection to file.

Best practice is:
```
with open(file, 'r') as f:
    content = f.read()
```

File: scanner/generate_byte_code.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+@click.command()
+@click.option('-t', type=click.Choice(['bytecode', 'abi']), help="The flag define which type you need.")
+@click.option('--solidity-file', '-s', help="The flag define path to solidity file.")
+@click.option('--contract', '-c', help="The flag define which contract must be convert.")
+def main(solidity_file, t, contract):
+    if t == 'bytecode':
+        if not contract:
+            get_contract_names(solidity_file)
+        else:
```
Comment (s0b0lev) [inline scanner/generate_byte_code.py]: Line 38 change to `return get_contract_names(solidity_file)`
In that case you can remove list 39 and `return convert_to_byte_code(solidity_file, contract)`

File: scanner/generate_byte_code.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                                                                                                 str(contract_name)]
+                                                                                                ['abi']}
+        print(report)
+
+
+def get_contract_names(file):
+    path_to_file = open(file, mode='r')
+    compiled_sol = compile_source(path_to_file.read())
+
+    name_of_contract = compiled_sol['<stdin>:SafeMath']['ast']['attributes']['exportedSymbols']
```
Comment (s0b0lev) [inline scanner/generate_byte_code.py]: Looks strange that we hardcode **SafeMath** here: `compiled_sol['<stdin>:SafeMath']`

File: tests/test_generate_byte_code.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import unittest
+from scanner.generate_byte_code import *
+
+
+class TestMethodsOfGenerate(unittest.TestCase):
+
+    def test_convert_to_byte_code(self):
+        file_to_assert = open('assert_with_name_test.txt', mode='r').read()
```
Comment (s0b0lev) [inline tests/test_generate_byte_code.py]: Also use:
```
with open(filename, 'r') as f:
    content = f.read()
```

**Comment by ersul4ik** [comment]
@s0b0lev fixed all needed corrections

**Pull request merged**